### PR TITLE
Prefer POSIX character classification for POSIX syntax

### PR DESCRIPTION
**Pull request by hamishknight**
When written as e.g `[:alpha:]` rather than `\p{alpha}`, classify as POSIX properties rather than Unicode.

### CODE REVIEW

**Comment by hamishknight** [comment]
@swift-ci please test Linux

File: Tests/RegexTests/ParseTests.swift [hunk 1]
```diff
@@ -174,12 +174,20 @@ extension RegexTests {
     )))
 
     parseTest("[[:alnum:]]", charClass(posixProp_m(.posix(.alnum))))
+    parseTest("[[:ascii:]]", charClass(posixProp_m(.posix(.alnum))))
```
Comment (kylemacomber) [inline Tests/RegexTests/ParseTests.swift:177]:
```suggestion
    parseTest("[[:ascii:]]", charClass(posixProp_m(.posix(. ascii))))
```

File: Tests/RegexTests/ParseTests.swift [hunk 2]
```diff
@@ -174,12 +174,20 @@ extension RegexTests {
...
     parseTest("[[:blank:]]", charClass(posixProp_m(.posix(.blank))))
+    parseTest("[[:cntrl:]]", charClass(posixProp_m(.posix(.cntrl))))
+    parseTest("[[:digit:]]", charClass(posixProp_m(.posix(.digit))))
     parseTest("[[:graph:]]", charClass(posixProp_m(.posix(.graph))))
+    parseTest("[[:lower:]]", charClass(posixProp_m(.posix(.lower))))
     parseTest("[[:print:]]", charClass(posixProp_m(.posix(.print))))
+    parseTest("[[:punct:]]", charClass(posixProp_m(.posix(.punct))))
+    parseTest("[[:space:]]", charClass(posixProp_m(.posix(.space))))
+    parseTest("[[:upper:]]", charClass(posixProp_m(.posix(.space))))
```
Comment (kylemacomber) [inline Tests/RegexTests/ParseTests.swift:186]:
```suggestion
    parseTest("[[:upper:]]", charClass(posixProp_m(.posix(. upper))))
```

File: Tests/RegexTests/ParseTests.swift (hunk 1 above)
Comment (hamishknight) [reply Tests/RegexTests/ParseTests.swift:177]: Ack, good catch! I actually realized last night that these tests don't correctly detect failures as it's comparing dump output which we haven't completely filled out yet 😬

**Comment by milseman** [comment]
Sorry for the red herring yesterday, I didn't realize that everything inside \p{...} was accepted in posix style syntax. I say we hold off for a little bit until we understand how options work

**Pull request closed without merging**